### PR TITLE
feat(theme): toggle

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
According to the guidelines, `toggle` has changed 

@mloussot Some instructions are still missing:
- [ ] inner shadow details 
- [x] focus / hover states
- [x] checked but disabled style

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR
https://jira.talendforge.org/browse/TSCH-47
http://guidelines.talend.com/document/92132#/ui-controls/switch-toggle

**[ ] This PR introduces a breaking change**

screen shot 2017-10-11 at 16 53 01

### CODE REVIEW

File: packages/theme/src/theme/_forms.switch.scss [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		+ *:after,
+		&:checked + *:before,
+		&:checked + *:after {
+			position: absolute;
+			content: '';
+			top: 0;
+			left: 0;
+			transition: background 0.2s ease, transform 0.2s ease;
+			border-radius: 10rem;
+			border-color: transparent;
```
Comment (jsomsanith) [inline packages/theme/src/theme/_forms.switch.scss]: This seems to be only on `:after` (on `:before` we have no border)

File: packages/theme/src/theme/_forms.switch.scss [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			box-shadow: inset 0.1rem 0.1rem 0.5rem $silver;
```
Comment (jsomsanith) [inline packages/theme/src/theme/_forms.switch.scss]: This seems to be only on :before (on :after overrides it)

File: packages/theme/src/theme/_forms.switch.scss (hunk 1 above)
Comment (frassinier) [reply packages/theme/src/theme/_forms.switch.scss]: thanks, I will fix it

File: packages/theme/src/theme/_forms.switch.scss (hunk 2 above)
Comment (frassinier) [reply packages/theme/src/theme/_forms.switch.scss]: thanks, I will fix it

File: packages/theme/src/theme/_forms.switch.scss [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			+ *:after {
+				border-color: transparent;
+			}
+		}
+
+		&:hover,
+		&:focus {
+			+ *:after {
+				background-color: $white;
+				box-shadow: 0 0.1rem 0.3rem 0.1rem rgba(150, 150, 150, 0.45)
```
Comment (ncomont) [inline packages/theme/src/theme/_forms.switch.scss]: missing semicolon

**Comment by mloussot** [comment]
Here is the complete box-shadow / inner-shadows descriptions : https://company-57688.frontify.com/document/92132#/ui-controls/switch-toggle

**Pull request merged**